### ISSUE TITLE
root_elements raises AttributeError on schema with XsdAnyElement

### ISSUE THREAD

**Issue by pythonator** [brunato/xmlschema#107]
Calling `root_elements` fails for schemas containing an element with no type.

This code
```python
import xmlschema
schema = xmlschema.XMLSchema(schema_text)
schema.root_elements
```
raises
```
...
C:\Anaconda3\lib\site-packages\xmlschema\validators\schema.py in root_elements(self)
    520                     if e is xsd_element:
    521                         continue
--> 522                     elif e.ref or e.is_global:
    523                         if e.name in names:
    524                             names.discard(e.name)

AttributeError: 'XsdAnyElement' object has no attribute 'ref'
```

for the schema
```xml
<?xml version="1.0" encoding="utf-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
  <xs:element name="root1" type="root"/>
  <xs:element name="root2" type="root"/>
  <xs:complexType name="root">
    <xs:sequence>
      <xs:element name="elementWithNoType"/>
    </xs:sequence>
  </xs:complexType>
</xs:schema>
```

Besides this line, there is three other places in the code that access the `ref` of an element (that isn't `self`). Those might also be problematic.

Would the proper fix be to add `ref = None` to the definition of `XsdAnyElement`?

**Comment by brunato** [comment]
> Would the proper fix be to add ref = None to the definition of XsdAnyElement?

It's better to avoid adding *ref* to XsdAnyElement, so I've added `or isinstance(e, XsdAnyElement)` to line 520:
```python
if e is xsd_element or isinstance(e, XsdAnyElement):
    continue
elif e.ref or e.is_global:
    if e.name in names:
        names.discard(e.name)
```
I will also put your sample schema into the test code. The fix will be available in the next release.
Thanks

**Comment by pythonator** [comment]
Ok, thanks!

**Comment by brunato** [comment]
Fix available with new release v1.0.13.
Bye
